upgrade_analysis/models/upgrade_analysis.py

```diff
@@ -0,0 +1,227 @@
+# Copyright 2011-2015 Therp BV <https://therp.nl>
+# Copyright 2016 Opener B.V. <https://opener.am>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+# flake8: noqa: C901
+
+import logging
+import os
+
+from odoo import fields, models
+from odoo.modules import get_module_path
+from odoo.tools import config
+
+from .. import compare
+
+_logger = logging.getLogger(__name__)
+_IGNORE_MODULES = ["openupgrade_records", "upgrade_analysis"]
+
+
+class UpgradeAnalysis(models.Model):
+    _name = "upgrade.analysis"
+    _description = "Upgrade Analyses"
+
+    analysis_date = fields.Datetime(readonly=True)
+
+    state = fields.Selection(
+        [("draft", "draft"), ("done", "Done")], readonly=True, default="draft"
+    )
+    config_id = fields.Many2one(
+        string="Comparison Config",
+        comodel_name="upgrade.comparison.config",
+        readonly=True,
+        required=True,
+    )
+
+    log = fields.Text(readonly=True)
+    upgrade_path = fields.Char(
+        default=config.get("upgrade_path", False),
+        help="The base file path to save the analyse files of Odoo modules",
+    )
```

Comment:
I don't understand why this is a field. Is it really possible to tell Odoo different upgrade paths per module? If not, I'd remove it and just get the info from the config.

Reply:
> I don't understand why this is a field. Is it really possible to tell Odoo different upgrade paths per module

yes it is. even it is very theoritical.

Reply:
I introduced it before I understood --upgrade-path is a standard Odoo option now. I think we can remove this now and  then we can reinstate apriori as a python module as far as I am concerned.